wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/ScatterElementToGraphNode.scala

```diff
@@ -25,16 +25,23 @@ import wom.graph.GraphNodePort.{ConnectedInputPort, InputPort, OutputPort}
 import wom.graph._
 import wom.graph.expression.{AnonymousExpressionNode, PlainAnonymousExpressionNode}
 import wom.types.{WomAnyType, WomArrayType, WomType}
+import wom.util.WomParseConfig
 
 object ScatterElementToGraphNode {
   def convert(a: ScatterNodeMakerInputs)
              (implicit expressionValueConsumer: ExpressionValueConsumer[ExpressionElement],
               fileEvaluator: FileEvaluator[ExpressionElement],
               typeEvaluator: TypeEvaluator[ExpressionElement],
               valueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[Set[GraphNode]] =
-    if (a.insideAnotherScatter) {
-      convertInnerScatter(a)
+    if (WomParseConfig.convertNestedScatterToSubworkflow) {
```

Comment:
I think it is a bit of a structural misstep to reference the config directly where it's being used. This is the primary reason why you are finding it hard to test with a different config value - there is no way to provide a mock config.

It would make our lives a lot easier if we applied a dependency injection pattern so that the value of `convertNestedScatterToSubworkflow` could be varied in tests. One nearby example is the `insideAScatter` which becomes a member of `GraphLikeConvertInputs`.

The functional code for this PR looks great and definitely works, but I think it's important to get the factoring of the config value right, for our collective sanity.

Reply:
Possibly the right `*Inputs` file to target would be `wdl.transforms.base.wdlom2wom.FileElementToWomBundleInputs` and then alternate `true` and `false` when we construct instances of it in `wdl.draft3.transforms.wdlom2wom.WdlFileToWomSpec`